src/test/java/com/flightstats/hub/dao/aws/DynamoChannelConfigDaoTest.java

```diff
@@ -35,4 +69,54 @@ void testSimple() {
         logger.info("channel {}", testSimple);
         assertNotNull(testSimple);
     }
+
+    @Test
+    public void testSingleGetReturnsNullIfConfigIsUnparseable() {
+        GetItemResult result = mock(GetItemResult.class);
+        when(result.getItem()).thenReturn(createBogusEntry());
+        when(dbClient.getItem(any(GetItemRequest.class))).thenReturn(result);
+        assertNull(mockedDao.get("bob"));
+    }
+
+    @Test
+    public void testListGetDropsUnparseableConfigs() {
+        ScanResult result = mock(ScanResult.class);
+
+        Map<String, AttributeValue> channel1 = toDynamoEntry("channel1", "Channel One", 1557433692138L);
+        Map<String, AttributeValue> channel2 = toDynamoEntry("channel2", "Channel Two", 1557433692138L);
+
+        Map<String,AttributeValue> bogusRecord = createBogusEntry();
+
+        when(result.getItems()).thenReturn(
+                Arrays.asList(
+                        channel1,
+                        bogusRecord,
+                        channel2
+                )
+        );
+        when(result.getLastEvaluatedKey()).thenReturn(null);
+        when(dbClient.scan(any(ScanRequest.class))).thenReturn(result);
+        Collection<ChannelConfig> cfgs = mockedDao.getAll(false);
+        assertNotNull(cfgs);
+        assertEquals(2, cfgs.size());
+        assertThat(cfgs, hasItems(
+                channelConfigDao.mapItem(channel1).get(),
+                channelConfigDao.mapItem(channel2).get()
+        ));
+    }
+
+    private Map<String,AttributeValue> toDynamoEntry(String key, String displayName, long date) {
+        Map<String,AttributeValue> channel = new HashMap<>();
+        channel.put("key", new AttributeValue(key));
+        channel.put("displayName", new AttributeValue(displayName));
+        channel.put("date", new AttributeValue().withN(String.valueOf(date)));
+        return channel;
+    }
+
+    private Map<String,AttributeValue> createBogusEntry() {
+        Map<String,AttributeValue> bogusRecord = new HashMap<>();
+        bogusRecord.put("i exist", new AttributeValue("to break your code"));
```

Comment:
does this work? Spaces in key does not look right.

Reply:
Yep.  At the end of the day, it's just a string.